test/services/SdesGatekeeperServiceSpec.scala

```diff
@@ -0,0 +1,217 @@
+
+class SdesGatekeeperServiceSpec extends SpecBase {
+
+  "SdesGatekeeperService" should {
+
+    "convert FileInformation to CashStatementFile correctly" in new Setup {
+      val sdesFileInformation: FileInformation = validCashStatementFileInformation
+
+      val result: CashStatementFile = sdesGatekeeperService.convertToCashStatementFile(sdesFileInformation)
+
+      result.filename mustBe sdesFileInformation.filename
+      result.downloadURL mustBe sdesFileInformation.downloadURL
+      result.metadata.periodStartYear mustBe periodStartYear
+      result.metadata.periodStartMonth mustBe periodStartMonth
+      result.metadata.periodStartDay mustBe periodStartDay
+      result.metadata.periodEndYear mustBe periodEndYear
+      result.metadata.periodEndMonth mustBe periodEndMonth
+      result.metadata.periodEndDay mustBe periodEndDay
+      result.metadata.fileFormat mustBe FileFormat(csv)
+      result.metadata.fileRole mustBe CashStatement
+      result.metadata.cashAccountNumber mustBe someAccountNumber
+      result.metadata.statementRequestId mustBe someRequestId
+    }
+
+    "convert FileInformation to VatCertificateFile correctly" in new Setup {
+      val sdesFileInformation: FileInformation = validVatCertificateFileInformation
+
+      val result: VatCertificateFile = sdesGatekeeperService.convertToVatCertificateFile(sdesFileInformation)
+
+      result.metadata.fileRole mustBe C79Certificate
+    }
+
+    "convert FileInformation to PostponedVatStatementFile correctly" in new Setup {
+      val sdesFileInformation: FileInformation = validPostponedVatStatementFileInformation
+
+      val result: PostponedVatStatementFile =
+        sdesGatekeeperService.convertToPostponedVatStatementFile(sdesFileInformation)
+
+      result.metadata.fileRole mustBe PostponedVATStatement
+      result.metadata.source mustBe "CDS"
+    }
+
+    "convert FileInformation to SecurityStatementFile correctly" in new Setup {
+      val sdesFileInformation: FileInformation = validSecurityStatementFileInformation
+
+      val result: SecurityStatementFile = sdesGatekeeperService.convertToSecurityStatementFile(sdesFileInformation)
+
+      result.metadata.fileRole mustBe SecurityStatement
+      result.metadata.checksum mustBe "MISSING CHECKSUM"
+    }
+
+    "convert FileInformation to DutyDefermentStatementFile correctly" in new Setup {
+      val sdesFileInformation: FileInformation = validDutyDefermentStatementFileInformation
+
+      val result: DutyDefermentStatementFile =
+        sdesGatekeeperService.convertToDutyDefermentStatementFile(sdesFileInformation)
+
+      result.metadata.fileRole mustBe DutyDefermentStatement
+      result.metadata.defermentStatementType mustBe DDStatementType("DefermentStatement")
+      result.metadata.dutyOverLimit mustBe Some(false)
+      result.metadata.dutyPaymentType mustBe Some("Unknown")
+      result.metadata.dan mustBe "Unknown"
+    }
+  }
+
+  "SdesGatekeeperService.convertTo" should {
```

Comment:
There should be no need to use the mock in this scenario. It should have been done by importing the implicits and then call the convertTo method but okay for moment.

Reply:
will be taken care later